xbmc/platform/posix/filesystem/SMBFile.cpp

```diff
@@ -29,14 +29,35 @@
 
 #include <inttypes.h>
 #include <mutex>
+#include <regex>
 
 #include <libsmbclient.h>
 
 using namespace XFILE;
 
-void xb_smbc_log(const char* msg)
+void xb_smbc_log(void* private_ptr, int level, const char* msg)
 {
-  CLog::Log(LOGINFO, "{}{}", "smb: ", msg);
+  const int logLevel = [level]()
+  {
+    switch (level)
+    {
+      case 0:
+        return LOGWARNING;
+      case 1:
+        return LOGINFO;
+      default:
+        return LOGDEBUG;
+    }
+  }();
+
+  if (strchr(msg, '@'))
```

Comment:
```suggestion
  if (std::strchr(msg, '@'))
```

Reply:
And include <cstring>

Reply:
Thank you.

It compiles without explicit include of `<cstring>`. It must have got included through some other include. Do you still prefer an explicit include to be added?

(I don't want to pollute the history with super-small commits, unless necessary. I also don't want to squash commits to respect authors/reviewers.)

Reply:
I've tried `squash` locally and it kept all the authors. I'll squash the commits then for cleaner history.

Reply:
> It compiles without explicit include of <cstring>. It must have got included through some other include. Do you still prefer an explicit include to be added?

Technically it is the most correct.

> (I don't want to pollute the history with super-small commits, unless necessary. I also don't want to squash commits to respect authors/reviewers.)

You can amend last commit